src/Server/MySQLHandler.cpp

```diff
@@ -66,7 +66,7 @@ static String showCountWarningsReplacementQuery(const String & query);
 static String selectEmptyReplacementQuery(const String & query);
 static String showTableStatusReplacementQuery(const String & query);
 static String killConnectionIdReplacementQuery(const String & query);
-static String selectLimitReplacementQuery(const String & query);
+static std::optional<String> setSettingReplacementQuery(const String & query, const String & mysql_setting, const String & native_setting);
```

Comment:
If you move the implementation of these methods before where they are used, you would not need to forward-declare them (l. 64-69).